proto/decentraland/social/friendships/friendships.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+package decentraland.social.friendships;
+
+import "google/protobuf/empty.proto";
+
+// Validation and anti-spam errors
+enum FriendshipErrorCode {
+  TOO_MANY_REQUESTS_SENT = 0;
+  NOT_ENOUGH_TIME_PASSED = 1;
+  BLOCKED_USER = 2;
+  NON_EXISTING_USER = 3;
+  INVALID_REQUEST = 4;
+  UNKNOWN = 5;
+}
+
+enum EventType {
+  REQUEST = 0;
+  ACCEPT = 1;
+  REJECT = 2;
+  CANCEL = 3;
+  DELETE = 4;
+}
+
+// This message is a response that is sent from the server to the client
+message FriendshipEvent {
+  oneof body {
+    Request request = 1;
+    Accept accept = 2;
+    Reject reject = 4;
+    Delete delete = 5;
+    Cancel cancel = 6;
+  }
+}
+
+message User { string address = 1; }
+
+message Users { repeated User users = 1; }
+
+message Request {
+  User user = 1;
+  int64 created_at = 2;
+  optional string message = 3;
+}
+
+message Requests {
+  int64 total = 1; // Total amount of friendship requests
+  repeated Request items = 2;
+}
+
+message RequestEvents {
+  Requests outgoing = 1; // Requests the authed user have sent to users
+  Requests incoming = 2; // Requests the authed user have received from users
+}
+
+message Accept { User user = 1; }
+
+message Reject { User user = 1; }
+
+message Delete { User user = 1; }
+
+message Cancel { User user = 1; }
+
+message GetFriendsPayload {
+  int32 limit = 1; // Max amount of entries to retrieve
+  int32 skip = 2;  // Amount of entries to skip
+  optional string search_term = 3;
+}
+
+message GetRequestEventsPayload {
+  int32 outgoing_limit = 1; // Max amount of entries of sent request events to retrieve
+  int32 outgoing_skip = 2; // Amount of entries of sent request events to skip
+  int32 incoming_limit = 3; // Max amount of entries of received request events to retrieve
+  int32 incoming_skip = 4; // Amount of entries of received request events to skip
+}
+
+// We expect new payloads in the future here, since each event type shall have
+// its own
+message UpdateFriendshipPayload {
+  User user = 1;
+  EventType event = 2;
+
+  oneof payload { Request request = 3; }
+}
+
+message UpdateFriendshipResponse {
+  oneof response {
+    FriendshipErrorCode error = 1;
+    FriendshipEvent event = 2;
+  }
+}
+
+message SubscribeFriendshipEventsUpdatesResponse {
+  repeated FriendshipEvent events = 1;
+}
+
+service FriendshipsService {
+  // Get the list of friends for the authenticated user
+  rpc GetFriends(GetFriendsPayload) returns (Users) {}
```

Comment:
Should the response show the limit and offset too? (The same question about request events)